linera-core/src/remote_node.rs

```diff
@@ -82,6 +82,29 @@ impl<N: ValidatorNode> RemoteNode<N> {
         self.check_and_return_info(response, chain_id)
     }
 
+    pub(crate) async fn handle_optimized_certificate(
+        &mut self,
+        certificate: &Certificate,
+        delivery: CrossChainMessageDelivery,
+    ) -> Result<Box<ChainInfo>, NodeError> {
+        if certificate.is_signed_by(&self.name) {
```

Comment:
What difference does it make to have this special handling for a certificate signed by the validator itself?

Reply:
The idea is that if the validator signed the certificate, they probably already have its value in the cache, so we don't need to send it.

Reply:
I thought that _this_ logic is already on the remote node (validator) but I guess it's a wrapper to communicate with it...

Reply:
Exactly, yes. Not sure what would be a better name. :thinking:

Reply:
Oh, maybe we shouldn't even send the certificate to a validator that signed it?